### PR TITLE
Change Legacy Hook with new Hook dispatche event: actionAdminProducts…

### PR DESCRIPTION
**Pull request by luigimassa**
Upgrade from Legacy Hook::exec to new Hook dispatcher

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop / 1.7.8.x
| Description?      | Upgrade from Legacy Hook::exec to new Hook dispatcher
| Type?             |  improvement 
| Category?         | BO
| BC breaks?        |  no
| Deprecations?     |  no
| Fixed ticket?     | Fix 28680
| Related PRs       |
| How to test?      | BO > Product > List product
| Possible impacts? |

### CODE REVIEW

File: src/Adapter/Product/AdminProductDataProvider.php [hunk 1]
```diff
@@ -52,6 +53,11 @@
  */
 class AdminProductDataProvider extends AbstractAdminQueryBuilder implements ProductInterface
 {
+    /**
+     * @var HookDispatcherInterface
+     */
+    protected $hookDispatcher;
```
Comment (matks) [inline src/Adapter/Product/AdminProductDataProvider.php:59]: Hello, as mentioned by PHPStan I think you need to add a `use` statement

**Review by matks — CHANGES_REQUESTED** [review]

File: src/Adapter/Product/AdminProductDataProvider.php (hunk 1 above)
Comment (luigimassa) [reply src/Adapter/Product/AdminProductDataProvider.php:59]: ops... fixed. Thanks

File: src/PrestaShopBundle/Resources/config/services/adapter/product.yml [hunk 2]
```diff
@@ -496,4 +497,4 @@ services:
     arguments:
       - '@prestashop.adapter.product.repository.product_multi_shop_repository'
       - '@prestashop.adapter.product.stock.repository.stock_available_multi_shop_repository'
-      - '@prestashop.adapter.shop.repository.shop_repository'
+      - '@prestashop.adapter.shop.repository.shop_repository'
```
Comment (matthieu-rolland) [inline src/PrestaShopBundle/Resources/config/services/adapter/product.yml]: apparently this file is missing an empty line at the end

**Review by matthieu-rolland — COMMENTED** [review]
LGTM 👍 , just a small feedback

File: src/PrestaShopBundle/Resources/config/services/adapter/product.yml (hunk 2 above)
Comment (luigimassa) [reply src/PrestaShopBundle/Resources/config/services/adapter/product.yml]: fixed

**Comment by matks** [comment]
I close and re-open to reset the CI. Sorry for late answer

**Comment by kpodemski** [comment]
Hello @luigimassa 

You have a merge commit, and we cannot accept it. Could you rebase Pull Request to keep the GIT history clean?

**Comment by luigimassa** [comment]
I've problem to rebase

**Comment by luigimassa** [comment]
Please check new pull request   #29521
excuse me and thanks

**Pull request closed without merging**